engine/src/main/java/org/terasology/i18n/TranslationUri.java

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright 2015 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.terasology.i18n;
+
+import java.util.Locale;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import org.terasology.engine.SimpleUri;
+import org.terasology.engine.Uri;
+import org.terasology.module.sandbox.API;
+import org.terasology.naming.Name;
+
+/**
+ * A URI to identify internationalization objects in Terasology.
+ * These URIs are always in the form: {@literal <module-name>:<project-name>#<id>}.
+ */
+@API
+public class TranslationUri implements Uri {
```

Comment:
Do we really need another Uri class? I've been trying to get rid of them. Won't AssetUri do?